### DISCUSSION TITLE
[2.6] Request goes on forever when using wire:model

### DISCUSSION THREAD

**Discussion by mabdullahsari** [livewire/livewire#3741] [7. Bugs (archived)] [answered]
A request that worked fine in 2.5, now enters (most probably) an infinite loop. I'm certainly glad to provide more info if needed, but am not really sure how to describe it in another way that is useful for you to reproduce and debug. 

The element in question:

```html
<input type="checkbox" name="property_group_1_10" value="10" id="10" wire:key="property_group_1_10" wire:model="filters.property_group_1">
```

The request payload that's being sent:

https://pastebin.com/Q2sJ6FLR

The field contents that's being attempted to be written to:

```php
array:7 [
    "colors" => []
    "property_group_2" => []
    "property_group_3" => []
    "property_group_1" => []
    "property_group_5" => []
    "brands" => []
    "property_group_14" => []
]
```

The issue:

```bash
[2021-09-15 10:57:10] local.ERROR: Maximum execution time of 30 seconds exceeded {"userId":3,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Maximum execution time of 30 seconds exceeded at /Users/masari/Sites/REDACTED/app/vendor/livewire/livewire/src/Component.php:111)
[stacktrace]
#0 {main}
"} 
```

**Comment by joshhanley** [comment]
@mabdullahsari it's probably an unrelated issue, but we just fixed a few other issues and tagged v2.6.2. 

Can you test on that version and confirm if your issue is still happening?

**Reply by mabdullahsari** [reply]
Unfortunately, it's still happening in v2.6.2.

**Comment by bangnokia** [comment]
Can confirm, I got the same error

**Comment by joshhanley** [comment]
@mabdullahsari or @bangnokia I can't replicate the issue, would either one of you be able to create a small demo repo that shows the issue so I can look into it? 🙂

Thanks!

**Reply by mabdullahsari** [reply]
Yeah, sure. I'll post a link here when it's up.

**Reply by joshhanley** [reply]
Thanks!

**Reply by mabdullahsari** [reply]
Sorry, haven't had the time to set up a demo repo. Seems like the fella below me got to the rescue. Hope you can smash that bug :)

**Comment by oxytoxin** [comment]
I also got this issue on my select inputs with wire:model using v2.6.3. I reverted to 2.5.5 for the time being. The stacktrace in my app is the same as in the OP.

**Reply by joshhanley** [reply]
@oxytoxin any chance you can provide some code that demonstrates the issue? 

I'm happy to get it fixed, but I can't seem to replicate the problem.

Thanks!

**Reply by oxytoxin** [reply]
@joshhanley Hey man! I managed to reproduce the issue and pushed a repo for it here:
https://github.com/oxytoxin/livewiretest

**Reply by joshhanley** [reply]
@oxytoxin legend! Thanks so much 🙂

**Comment by joshhanley** [comment]
Thanks to @oxytoxin's repo, I know what's going on here. Now I just have to work out the best way to fix it haha.

What is happening is, we have changed the pagination, so that whenever the pagination properties change, they fire off hooks, the same as other properties do.

It allows us to a have `updatingPage` and `updatedPage` methods that get fired, so people can do things like fire browser events when the page changes, allowing them to setup things like "scroll to top", or whatever they need.

But if you have `$this->resetPage` in a generic hook like `updating/updated` then when reset is called, it changes the pagination property. This triggering all the hooks, which includes `updating/updated`, which in turn is calling `$this->resetPage` again, causing an infinite loop.

Example code that causes issues

```php
public function updating() {
    $this->resetPage();
}
```

I'll have to have a think about that, but if anyone has any suggestions, I'm open to them 🙂

**Comment by joshhanley** [comment, marked as answer]
This has been fixed and tagged in v2.6.5. 🙂

@mabdullahsari or @oxytoxin can you please test and let me know if there are any more issues? 

Thanks!

**Reply by oxytoxin** [reply]
You rock man! Issue has been fixed both on my test repo and in the project I have been working on. Thanks a bunch!

**Reply by joshhanley** [reply]
@oxytoxin thanks for testing, glad to hear it worked! No worries 🙂
